src/mjolnir/linkclassification.cc

```diff
@@ -439,6 +423,311 @@ struct LinkGraphBuilder {
   }
 };
 
+#ifdef LOGGING_LEVEL_DEBUG
+json::MapPtr
+LineStringFeature(Data& data, const std::vector<uint32_t>& nodes, std::string color = "") {
+  auto feature = json::map({});
+  feature->emplace("type", std::string("Feature"));
+  auto properties = json::map({});
+  if (color.size()) {
+    properties->emplace("stroke", color);
+  }
+  feature->emplace("properties", properties);
+  {
+    auto geojson = json::map({});
+    auto coords = json::array({});
+    coords->reserve(nodes.size());
+    for (auto i : nodes) {
+      auto p = (*data.nodes[i]).node.latlng();
+      coords->emplace_back(json::array({json::fixed_t{p.lng(), 6}, json::fixed_t{p.lat(), 6}}));
+    }
+    geojson->emplace("type", std::string("LineString"));
+    geojson->emplace("coordinates", std::move(coords));
+
+    feature->emplace("geometry", std::move(geojson));
+  }
+
+  return feature;
+}
+
+json::MapPtr PointFeature(Data& data, uint32_t node, std::string color = "") {
+  auto feature = json::map({});
+  feature->emplace("type", std::string("Feature"));
+  auto properties = json::map({});
+  if (color.size()) {
+    properties->emplace("marker-color", color);
+  }
+  feature->emplace("properties", properties);
+  {
+    auto geometry = json::map({});
+    geometry->emplace("type", std::string("Point"));
+
+    auto p = (*data.nodes[node]).node.latlng();
+    auto coords = json::array({json::fixed_t{p.lng(), 6}, json::fixed_t{p.lat(), 6}});
+    geometry->emplace("coordinates", std::move(coords));
+
+    feature->emplace("geometry", std::move(geometry));
+  }
+
+  return feature;
+}
+
+std::string VisualizeIntersection(
```

Comment:
helpful function to visualize slip lane result 
![Screenshot 2021-07-08 at 18 03 25](https://user-images.githubusercontent.com/14297128/124945439-d016f580-e016-11eb-9690-e2bcbb6c1a33.png)